runtime/win32.c

```diff
@@ -1113,54 +1104,12 @@ void caml_init_os_params(void)
   QueryPerformanceFrequency(&frequency);
   /* Convert the frequency to the duration of 1 tick in ns */
   frequency.QuadPart = 1000000000LL / frequency.QuadPart;
```

Comment:
What is the expected range of values for `frequency`?  How much error do we get by computing `x * (1000000000 / y)` using integer arithmetic ?  What about floating-point arithmetic ?

Reply:
I don't want to give the false impression that this was hugely thought through, but when this was added in 2018 (first version of the Windows multicore port), my memory is that it was more important that the clock source was _strictly_ monotonic than that it was strictly accurate - although a system with a bad counter (as low as 1us) would still take 128 days' uptime before QueryPerformanceCounter * (1e9 / QueryPerformanceFrequency) exceeds 2^53. I think it stayed with integer arithmetic because it required less thought about whether it was strictly monotonic!

As it happens, on both the systems I immediately had to hand, the frequency is 1e7, so no loss of precision for lucky users of 3990Xs or whatever i7 my laptop has in it, but a few other systems have somewhat less regular values.